core/psh/psh.c

```diff
@@ -92,8 +93,9 @@ int main(int argc, char **argv)
 		usleep(50000);
 
 	/* Check if its first shell */
-	tcpid = tcgetpgrp(STDIN_FILENO);
+	psh_common.tcpid = tcgetpgrp(STDIN_FILENO);
 	base = basename(argv[0]);
+	ispshlogin = (strcmp(base, "pshlogin") == 0);
 	do {
 		/* login prompt */
 		if (strcmp(base, "pshlogin") == 0 && (app = psh_findapp("auth")) != NULL)
```

Comment:
We already did strcmp ;) Use `ispshlogin`

Reply:
Oh. Shame I missed that. Thanks for pointing it out!